src/app/Dashboard/AutomatedAnalysis/AutomatedAnalysisCard.tsx

```diff
@@ -0,0 +1,724 @@
+
+  const handleArchivedRecordings = React.useCallback(
+    (recordings: ArchivedRecording[]) => {
+      const freshestRecording = recordings.reduce((prev, current) =>
+        prev?.archivedTime > current?.archivedTime ? prev : current
+      );
+      addSubscription(
+        context.target.target().subscribe((target) => {
+          context.reports
+            .reportJson(freshestRecording, target.connectUrl)
+            .pipe(first())
+            .subscribe({
+              next: (report) => {
+                setUsingArchivedReport(true);
+                setReportTime(freshestRecording.archivedTime);
+                categorizeEvaluation(report);
+              },
+              error: () => {
+                handleStateErrors(INTERNAL_ERROR_MESSAGE);
+              },
+            });
+        })
+      );
+    },
+    [
+      addSubscription,
+      context.target,
+      context.reports,
+      categorizeEvaluation,
+      handleStateErrors,
+      setUsingArchivedReport,
+      setReportTime,
+    ]
+  );
+
+  const handleEmptyRecordings = React.useCallback(
+    (connectUrl: string) => {
+      const cachedReportAnalysis = context.reports.getCachedAnalysisReport(connectUrl);
+      if (cachedReportAnalysis.report.length > 0) {
+        setUsingCachedReport(true);
+        setReportTime(cachedReportAnalysis.timestamp);
+        categorizeEvaluation(cachedReportAnalysis.report);
+      } else {
+        addSubscription(
+          queryArchivedRecordings(connectUrl)
+            .pipe(
+              first(),
+              map((v) => v.data.archivedRecordings.data as ArchivedRecording[])
+            )
+            .subscribe({
+              next: (recordings) => {
+                if (recordings.length > 0) {
+                  handleArchivedRecordings(recordings);
+                } else {
+                  handleStateErrors(NO_RECORDINGS_MESSAGE);
+                }
+              },
+              error: () => {
+                handleStateErrors(NO_RECORDINGS_MESSAGE);
+              },
+            })
+        );
+      }
+    },
+    [
+      addSubscription,
+      context.reports,
+      categorizeEvaluation,
+      queryArchivedRecordings,
+      handleArchivedRecordings,
+      handleStateErrors,
+      setUsingCachedReport,
+      setReportTime,
+    ]
+  );
+
+  const generateReport = React.useCallback(() => {
+    addSubscription(
+      context.target.target().subscribe((target) => {
+        handleLoading();
+        setTargetConnectURL(target.connectUrl);
+        dispatch(automatedAnalysisAddTargetIntent(target.connectUrl));
+        // query for designated automated-analysis profiling recording
+        addSubscription(
+          queryActiveRecordings(target.connectUrl)
+            .pipe(
+              first(),
+              map((v) => v.data.targetNodes[0].recordings.active.data[0] as Recording),
+              tap((recording) => {
+                if (recording == null) {
```

Comment:
We should check `undefined` since out-of-bound access will return undefined. Or we can just switch to check if the array of recording is undefined or length == 0